sql/sql_list.cc

```diff
@@ -23,13 +23,6 @@
 
 list_node end_of_list;
 
-void free_list(I_List <i_string_pair> *list)
```

Comment:
Instead of removing `free_list` and creating your own version in `rpl_filter.cc`; you can just add a destructor for the class `i_string_pair` which frees `key` and `val`.

Reply:
Well it was written to be re-used elsewhere (although it never was). If you want to delete it and move it only to where it is used (the `rpl_filter` namespace), I would ask to make it self-contained. I.e., also move the definition of `i_string_pair` into `rpl_filter.h`, change the name `i_string_pair` to no longer imply its use as generic, but something that construes that it is used for rewrite_db logic (e.g. `rewrite_db_rule` ), etc. Even in doing this, I would suggest the destructor pattern and keeping the body of `free_list` as is (although the function name `free_list` should be changed).